web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/ViewSettingController.java

```diff
@@ -66,9 +75,66 @@ public ResponseEntity<Object> getViewSetting(@PathVariable("id") String id) {
         return new ResponseEntity<>(result, HttpStatus.OK);
     }
 
+    @PutMapping(value = "/edit/viewSettings", produces = APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public ResponseEntity<Object> updateViewSetting(@RequestParam Map<String,String> allParams) {
+        Map<String, Object> result = new HashMap<>();
+
+        if (badParams(allParams)) {
+            result.put("error", "Request must include ids and view settings");
+            return new ResponseEntity<>(result, HttpStatus.BAD_REQUEST);
+        }
+
+        // see if view behavior is valid
+        var viewBehavior = caseInsensitiveValueOf(allParams.get("behavior"));
+
+        var ids = allParams.get("targets").split(",");
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        for (String id : ids) {
+            var pid = PIDs.get(id);
+            // check permissions for object first
+            accessControlService.assertHasAccess("Insufficient permissions to edit view settings for " + id,
+                    pid, principals, Permission.editViewSettings);
+
+            //check if object is a work
+            var repositoryObject = repositoryObjectLoader.getRepositoryObject(pid);
+            if (!(repositoryObject instanceof WorkObject)) {
+                throw new InvalidOperationForObjectType("Cannot update View Settings of object " +
+                        id + ", as only WorkObjects have View Settings");
+            }
+        }
+
+        // now build and send requests
+        for (String id : ids) {
+            var request = buildRequest(id, viewBehavior);
+            try {
+                viewSettingRequestSender.sendToQueue(request);
+            } catch (IOException e) {
+                log.error("Error updating view setting for {}", request.getObjectPidString(), e);
+                return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+
+        }
+
+        result.put("ids", ids);
+        result.put("status", "Submitted view setting updates for " + ids.length + " object(s)");
+        result.put("timestamp", System.currentTimeMillis());
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
     private String getValue(Resource resource, Property property) {
         var propValue = resource.getProperty(property);
         return propValue == null ? null : propValue.getString();
     }
 
+    private ViewSettingRequest buildRequest(String id, ViewSettingRequest.ViewBehavior viewBehavior) {
+        var request = new ViewSettingRequest();;
+        request.setObjectPidString(id);
+        request.setViewBehavior(viewBehavior);
+        return request;
+    }
+
+    private boolean badParams(Map<String,String> params) {
```

Comment:
In java, the naming convention for methods is to use a verb:
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html
so maybe `validateParams`, `checkForBadParams`, `hasBadParams`, etc.